532 custom ai feedback demo updates

andreash94 opened the pull request:
Adding info tab for tone, voice, level of detail and grade level. Showing grade for the graded assignment on assignment details page. Added ability to adjust for student grade level when prompting the LLM.

andreash94 committed:
adding info bar for Custom AI feedback

andreash94 committed:
adding grade percent detail to essay detail page

andreash94 committed:
added ability to adjust for grade levels

rappleb1 commented on LearningLens2025/frontend/lib/Views/view_submissions.dart:
```diff
@@ -494,6 +521,38 @@ class SubmissionListState extends State<SubmissionList> {
+                                                        'Middle School',
+                                                        'High School',
+                                                        'College'
+                                                      ]
```
There's a constant of these, please use it for consistency with other views - examples are in the analytics and rubric creation views.

andreash94 replied:
Looking at those learninglensconstants it's going to be more problematic longer term referencing that I think since it isn't a map, its a list. I would have to hard code the position. Also, having the available options listed in the actual view submissions is consistent with the other option choices as well.

rappleb1 replied:
I think you misunderstood what I meant here.
```suggestion
                                                      items: LearningLensConstants.gradeLevels
```

rappleb1 replied:
See analytics page as an example:
<img width="606" height="396" alt="image" src="https://github.com/user-attachments/assets/eef570e4-d2be-415c-85a2-1bed993f97ba" />

rappleb1 commented on LearningLens2025/frontend/lib/Views/view_submissions.dart:
```diff
@@ -494,6 +521,38 @@ class SubmissionListState extends State<SubmissionList> {
                                                     SizedBox(height: 4),
+
+                                                    // Grade Level Dropdown
+                                                    DropdownButtonFormField<
```
Is it possible to change these a bit? It is impossible to view a full "card" within the same scroll because there's not enough space.

<img width="587" height="432" alt="image" src="https://github.com/user-attachments/assets/0c0ac098-8ba2-4691-ab06-3c43880fe328" />

Maybe something like this with 2 items per line?
<img width="529" height="353" alt="image" src="https://github.com/user-attachments/assets/142bd2ba-44d7-47ea-bc39-18af097aab6a" />

andreash94 replied:
It is possible, but me attempting to change it is messing up other formatting and causing issues at the moment. In the interest of getting something stable for tonight in prep for Saturday I would just have this as a future fix.

rappleb1 committed:
layout fixes

rappleb1 reviewed (changes requested):
I am getting exceptions when clicking the buttons and using DeepSeek:
<img width="935" height="42" alt="image" src="https://github.com/user-attachments/assets/c922efbd-2d10-40aa-8328-d61bfde24548" />

Also pushed up some changes to fix the layout - longer essays wouldn't scroll and would cut text off so I shrunk that down and added a link to the full thing if the user clicks the header. Also converted out the HTML characters. (Decided to take care of it since it was unrelated to your PR but merging with another PR would likely be a pain since they touch the same files)

andreash94 commented:
> I am getting exceptions when clicking the buttons and using DeepSeek: <img alt="image" width="935" height="42" src="https://private-user-images.githubusercontent.com/43281004/504507498-c922efbd-2d10-40aa-8328-d61bfde24548.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyNjU3NDQsIm5iZiI6MTc2MTI2NTQ0NCwicGF0aCI6Ii80MzI4MTAwNC81MDQ1MDc0OTgtYzkyMmVmYmQtMmQxMC00MGFhLTgzMjgtZDYxYmZkZTI0NTQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDAwMjQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2ZmJmMzU3Y2MxNDhiNmQyODgxYmQwMWMxMTY5Nzk2NTExNjI3NDg3YjJlYTk1ZTI3YmNlN2IwZTZiZmUwODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9Hb06LqLLNGVxntrue9R6ov5EdvLMQL4hNNZFlvBCSM">
> 
> Also pushed up some changes to fix the layout - longer essays wouldn't scroll and would cut text off so I shrunk that down and added a link to the full thing if the user clicks the header. Also converted out the HTML characters. (Decided to take care of it since it was unrelated to your PR but merging with another PR would likely be a pain since they touch the same files)

@rappleb1 when are you seeing exceptions? I am unable to recreate.

rappleb1 commented:
> > I am getting exceptions when clicking the buttons and using DeepSeek: <img alt="image" width="935" height="42" src="https://private-user-images.githubusercontent.com/43281004/504507498-c922efbd-2d10-40aa-8328-d61bfde24548.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyNjU3NDQsIm5iZiI6MTc2MTI2NTQ0NCwicGF0aCI6Ii80MzI4MTAwNC81MDQ1MDc0OTgtYzkyMmVmYmQtMmQxMC00MGFhLTgzMjgtZDYxYmZkZTI0NTQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDI0VDAwMjQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2ZmJmMzU3Y2MxNDhiNmQyODgxYmQwMWMxMTY5Nzk2NTExNjI3NDg3YjJlYTk1ZTI3YmNlN2IwZTZiZmUwODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9Hb06LqLLNGVxntrue9R6ov5EdvLMQL4hNNZFlvBCSM">
> > Also pushed up some changes to fix the layout - longer essays wouldn't scroll and would cut text off so I shrunk that down and added a link to the full thing if the user clicks the header. Also converted out the HTML characters. (Decided to take care of it since it was unrelated to your PR but merging with another PR would likely be a pain since they touch the same files)
> 
> @rappleb1 when are you seeing exceptions? I am unable to recreate.

It might just be test data that got broken at some point? The "Rise and Fall of the Roman Empire" has no buttons that work, same as the bird essay and some others as well. But other essays seem to work fine? Not sure what is happening.

rappleb1 commented on LearningLens2025/frontend/lib/Views/view_submissions.dart:
```diff
@@ -494,6 +521,38 @@ class SubmissionListState extends State<SubmissionList> {
+                                                          gradeLevelSelectionMap[
+                                                                  participant
+                                                                      .id] ??
+                                                              'College',
```
```suggestion
                                                          gradeLevelSelectionMap[
                                                                  participant
                                                                      .id] ??
                                                        LearningLensConstants.gradeLevels.last,
```

rappleb1 committed:
fix for assignments outside of edulense

rappleb1 committed:
grade level

rappleb1 committed:
prompting

rappleb1 committed:
Merge remote-tracking branch 'origin/team_e' into 532_CustomAIFeedback_Demo_Updates

rappleb1 committed:
format

rappleb1 committed:
fix and format
